sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -932,7 +932,7 @@ object PushPredicateThroughJoin extends Rule[LogicalPlan] with PredicateHelper {
         split(joinCondition.map(splitConjunctivePredicates).getOrElse(Nil), left, right)
 
       joinType match {
-        case _: InnerLike | LeftExistence(_) =>
+        case _: InnerLike |  LeftSemi | ExistenceJoin(_) =>
```

Comment:
I should have included the full comment from the JIRA. The keyword is "this is not currently exposed". Here is the first part of my comment:

"`ExistenceJoin` should be treated the same as `LeftOuter` and `LeftAnti`, not `InnerLike` and `LeftSemi`. This is not currently exposed because the rewrite of `[NOT] EXISTS OR ...` to `ExistenceJoin` happens in rule `RewritePredicateSubquery`, which is in a separate rule set and placed after the rule `PushPredicateThroughJoin`. During the transformation in the rule `PushPredicateThroughJoin`, an `ExistenceJoin` never exists."